specs/core/1_beacon-chain-misc.md

```diff
@@ -0,0 +1,242 @@
+# Phase 1 miscellaneous beacon chain changes
+
+## Table of contents
+
+<!-- TOC -->
+
+- [Phase 1 miscellaneous beacon chain changes](#phase-1-miscellaneous-beacon-chain-changes)
+    - [Table of contents](#table-of-contents)
+    - [Configuration](#configuration)
+    - [Containers](#containers)
+        - [`CompactCommittee`](#compactcommittee)
+        - [`ShardReceiptProof`](#shardreceiptproof)
+    - [Helper functions](#helper-functions)
+        - [`pack_compact_validator`](#pack_compact_validator)
+        - [`unpack_compact_validator`](#unpack_compact_validator)
+        - [`committee_to_compact_committee`](#committee_to_compact_committee)
+        - [`get_previous_power_of_2`](#get_previous_power_of_2)
+        - [`verify_merkle_proof`](#verify_merkle_proof)
+        - [`compute_historical_state_generalized_index`](#compute_historical_state_generalized_index)
+        - [`get_generalized_index_of_crosslink_header`](#get_generalized_index_of_crosslink_header)
+        - [`process_shard_receipt_proof`](#process_shard_receipt_proof)
+    - [Changes](#changes)
+        - [Phase 0 container updates](#phase-0-container-updates)
+            - [`BeaconState`](#beaconstate)
+            - [`BeaconBlockBody`](#beaconblockbody)
+        - [Persistent committees](#persistent-committees)
+        - [Shard receipt processing](#shard-receipt-processing)
+
+<!-- /TOC -->
+
+## Configuration
+
+| Name | Value | Unit | Duration
+| - | - | - | - |
+| `MAX_SHARD_RECEIPT_PROOFS` | `2**0` (= 1) | - | - |
+| `PERIOD_COMMITTEE_ROOT_LENGTH` | `2**8` (= 256) | periods | ~9 months |
+| `MINOR_REWARD_QUOTIENT` | `2**8` (=256) | - | - |
+
+## Containers
+
+#### `CompactCommittee`
+
+```python
+class CompactCommittee(Container):
+    pubkeys: List[BLSPubkey, MAX_VALIDATORS_PER_COMMITTEE]
+    compact_validators: List[uint64, MAX_VALIDATORS_PER_COMMITTEE]
```

Comment:
What do you think about adding a custom type `CompactValidator: uint64`?

Reply:
> Remove unpack_compact_validator from 1_beacon-chain-misc.md since it's not used here.

Where would it be put though? In the light client file that actually uses it? I can see how that's theoretically better in one way, but it has a big disadvantage, which is that pack and unpack are no longer beside each other. So I'd still favor putting both in the same file, to make it easier for readers to see that they are inverses.